packages/contracts/contracts/L1/AssetManager.sol

```diff
@@ -392,10 +350,8 @@ contract AssetManager is ISemver, IERC721Receiver, IAssetManager {
             uint128 kroInKgh = KGH_MANAGER.totalKroInKgh(tokenIds[i]);
             uint128 kroSharesForTokenId = _convertToKroShares(validator, kroInKgh);
 
-            _vaults[validator].kghDelegators[msg.sender].shares[tokenIds[i]] = KghDelegatorShares({
-                kro: kroSharesForTokenId,
-                kgh: kghShares
-            });
+            KghDelegator storage kghDelegator = _vaults[validator].kghDelegators[msg.sender];
```

Comment:
How about declaring this before the for loop? I think it will reduce the gas slightly

Reply:
Just found that above comments of @seolaoh cover these, I'll resolve them.